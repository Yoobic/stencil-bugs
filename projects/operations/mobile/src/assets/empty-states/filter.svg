<svg xmlns="http://www.w3.org/2000/svg" xmlns:xlink="http://www.w3.org/1999/xlink" viewBox="0 0 135 135"><defs><style>.cls-1{fill:url(#linear-gradient);}.cls-2{fill:#e1dfff;}.cls-3{fill:#07ccc0;}.cls-4{fill:#3a3569;}.cls-5{fill:url(#linear-gradient-2);}.cls-6{fill:url(#linear-gradient-3);}.cls-7{fill:url(#linear-gradient-4);}</style><linearGradient id="linear-gradient" x1="22.4" y1="65.81" x2="114.9" y2="65.81" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#673fef"/><stop offset="0.61" stop-color="#877def"/><stop offset="1" stop-color="#5ab0ff"/></linearGradient><linearGradient id="linear-gradient-2" x1="31.38" y1="72.85" x2="43.32" y2="61.68" gradientUnits="userSpaceOnUse"><stop offset="0" stop-color="#ef6e7f"/><stop offset="1" stop-color="#f89c59"/></linearGradient><linearGradient id="linear-gradient-3" x1="58.06" y1="54.93" x2="70" y2="43.76" xlink:href="#linear-gradient-2"/><linearGradient id="linear-gradient-4" x1="84.47" y1="80.93" x2="96.41" y2="69.76" xlink:href="#linear-gradient-2"/></defs><title>Yoobic</title><g id="Filter"><path class="cls-1" d="M22.4,24.27l86.84-3.52a5.43,5.43,0,0,1,5.66,5.43v84.7H27.53a5.13,5.13,0,0,1-5.13-5.13V24.27Z"/><path class="cls-2" d="M20.1,27.65,107,24.13a5.43,5.43,0,0,1,5.65,5.43v84.69H25.23a5.13,5.13,0,0,1-5.13-5.13V27.65Z"/><circle class="cls-3" cx="36.91" cy="102.37" r="5.35"/><circle class="cls-3" cx="36.61" cy="102.37" r="5.35"/><circle class="cls-3" cx="63.29" cy="102.37" r="5.35"/><circle class="cls-3" cx="89.7" cy="102.37" r="5.35"/><path class="cls-4" d="M36.61,88.45a1,1,0,0,1-1-1V39.33a1,1,0,0,1,2,0V87.45A1,1,0,0,1,36.61,88.45Z"/><path class="cls-4" d="M63.29,88.45a1,1,0,0,1-1-1V39.33a1,1,0,0,1,2,0V87.45A1,1,0,0,1,63.29,88.45Z"/><path class="cls-4" d="M89.7,88.45a1,1,0,0,1-1-1V39.33a1,1,0,0,1,2,0V87.45A1,1,0,0,1,89.7,88.45Z"/><rect class="cls-5" x="28.09" y="63.7" width="17.04" height="8.5" rx="4.25" ry="4.25"/><rect class="cls-6" x="54.77" y="45.79" width="17.04" height="8.5" rx="4.25" ry="4.25"/><rect class="cls-7" x="81.18" y="71.79" width="17.04" height="8.5" rx="4.25" ry="4.25"/></g></svg>