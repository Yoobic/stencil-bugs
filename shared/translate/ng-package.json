{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
  "dest": "../../node_modules/@shared/translate",
  "deleteDestPath": false,
  "lib": {
    "entryFile": "src/public_api.ts",
    "umdModuleIds": {
      "lodash-es": "lodash-es",
      "moment": "moment",
      "rxjs": "rxjs",
      "@shared/common": "@shared/common",
      "@ngx-translate/core": "@ngx-translate/core",
      "@shared/interfaces": "@shared/interfaces"
    }
  }
}