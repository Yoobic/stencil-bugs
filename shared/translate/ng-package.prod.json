{
  "$schema": "../../node_modules/ng-packagr/ng-package.schema.json",
    "dest": "../../node_modules/@shared/translate",
  "lib": {
    "entryFile": "src/public_api.ts",
    "styleIncludePaths": [
      "../../../shared/styles"
    ]
  }
}