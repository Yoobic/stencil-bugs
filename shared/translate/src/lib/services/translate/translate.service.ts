import { Injectable } from '@angular/core';
import { CoreConfig, Persistent } from '@shared/common';
import { TranslateService, MissingTranslationHandler, MissingTranslationHandlerParams } from '@ngx-translate/core';
import { Observable, Subject } from 'rxjs';
export const AVAILABLE_LANGUAGES = [
    'en',
    'us',
    'fr',
    'de',
    'es',
    'nl',
    'pl',
    'it',
    'ru',
    'zhs',
    'zht',
    'pt',
    'kr',
    'ja',
    'ua',
    'he',
    'ar',
    'cz',
    'th',
    'tr',
    'bg',
    'el',
    'sl',
    'sk'
];

import { moment } from '@shared/interfaces';

@Injectable()
export class Translate {
    public static availablesLanguage: Array<string> = AVAILABLE_LANGUAGES;

    public static availablesLanguageAll: Array<string> = ['all'].concat(Translate.availablesLanguage);

    public static currentLanguage: string = '';

    public static loadExtraLanguage: (language) => Promise<any> = null;

    @Persistent('defaultLanguage') private defaultLanguage;

    private _languageChange = new Subject<string>();

    public static checkLanguage(language: string) {
        if (language === 'zh-Hant-HK' || language === 'zht') {
            language = 'zht';
        } else if (language === 'zh-Hans-GB' || language.startsWith('zh')) {
            language = 'zhs';
        } else {
            language = language.substring(0, 2);
        }
        if (Translate.availablesLanguage.indexOf(language) < 0) {
            language = 'en';
        }
        return language;
    }

    // public static loadFile(language: string) {
    //     language = Translate.checkLanguage(language);
    //     return System.import(/* webpackChunkName: "translate" */'../json/' + (language || 'en') + '.json').then(translations => {
    //         Translate[language] = translations;
    //         if (Translate.loadExtraLanguage) {
    //             return Translate.loadExtraLanguage(language);
    //         } else {
    //             return Promise.resolve({});
    //         }
    //     }).then(extras => {
    //         Translate[language] = Object.assign(Translate[language], extras || {});
    //         return Translate[language];
    //     });
    // }

    constructor(private ngTranslate: TranslateService, private coreConfig: CoreConfig) { }

    init(): Promise<void> {
        let browserLanguage = 'en';
        //this.ngTranslate.addLangs(AVAILABLE_LANGUAGES);
        if (!this.coreConfig.isUniversal()) {
            browserLanguage = (<any>window).navigator.userLanguage || window.navigator.language || 'en';
        }
        // if (this.coreConfig.isCordova() && !this.defaultLanguage) {
        //     return this.globalization.getPreferredLanguage().then(
        //         res => this.setLanguage(res.value),
        //         err => this.setLanguage(browserLanguage)
        //     );
        // } else {
        this.defaultLanguage = this.defaultLanguage || browserLanguage;
        return this.setLanguage(this.defaultLanguage);
        //        }
    }

    get languageChange$(): Observable<string> {
        return this._languageChange.asObservable();
    }

    setLanguage(language: string): Promise<any> {
        return new Promise((resolve, reject) => {
            if (language) {
                language = Translate.checkLanguage(language);
                this.defaultLanguage = language;
                this.ngTranslate.setDefaultLang(this.defaultLanguage);
                moment.locale(this.getLanguageFull(language, '-'));
                this._languageChange.next(language);
                Translate.currentLanguage = language;
                if (typeof window !== 'undefined' && window.document) {
                    if (this.isRtl(language)) {
                        window.document.dir = 'rtl';
                    } else {
                        window.document.dir = 'ltr';
                    }
                }
                this.ngTranslate.use(language).subscribe(() => {
                    resolve(language);
                });
            } else {
                resolve(null);
            }
        });
    }

    addDynamicLanguage(language: string, translations: Array<any>, isAdmin: boolean) {
        let newTranslations = (<any>this.ngTranslate).translations[language] || {};
        translations.forEach(d => {
            if (d && d.key && (!isAdmin || this.ngTranslate.instant(d.key) === d.key)) {
                newTranslations[d.key] = d[language];
            }
        });
        this.ngTranslate.setTranslation(language, newTranslations, true);
    }

    getCurrentLanguage() {
        return this.defaultLanguage;
    }

    getCurrentAngularLocale() {
        switch (this.defaultLanguage) {
            case 'en':
                return 'en-GB';
            case 'us':
                return 'en-US';
            case 'zhs':
                return 'zh-Hans';
            case 'zht':
                return 'zh-Hant';
            case 'fr':
            case 'de':
            case 'es':
            case 'nl':
            case 'pl':
            case 'it':
            case 'ru':
            case 'pt':
            case 'ja':
            case 'he':
            case 'ar':
            case 'th':
            case 'tr':
            case 'bg':
            case 'gr':
            case 'sl':
            case 'sk':
                return this.defaultLanguage;
            default:
                return 'en-US';
        }
    }

    getCurrentLanguageFull(separator = '_') {
        let l = this.getCurrentLanguage();
        return this.getLanguageFull(l, separator);
    }

    get(key: string, params?: Object): string {
        try {
            return this.ngTranslate.instant(key, params);
        } catch (err) {
            return key;
        }
    }

    getIcon(language: string) {
        switch (language) {
            case 'en':
                return 'flag-icon flag-icon-gb';
            case 'zhs':
                return 'flag-icon flag-icon-cn';
            case 'zht':
                return 'flag-icon flag-icon-hk';
            case 'ja':
                return 'flag-icon flag-icon-jp';
            case 'he':
                return 'flag-icon flag-icon-il';
            case 'ar':
                return 'flag-icon flag-icon-ae';
            case 'el':
                return 'flag-icon flag-icon-gr';
            case 'sl':
                return 'flag-icon flag-icon-si';
            default:
                return 'flag-icon flag-icon-' + language;
        }
    }

    getAll(language: string) {
        return Translate[language];
    }

    // loadAllFiles() {
    //     return Promise.all(Translate.availablesLanguage.map(l => Translate.loadFile(l)));
    // }

    polyglot(value: string) {
        if (value && value.replace && value.indexOf('%') >= 0) {
            return value.replace(/%(\w+)%/g, (match, key) => {
                return this.get(key);
            });
        }
        if (value && value.toUpperCase && value.toUpperCase() === value) {
            return this.get(value);
        }
        return value;
    }

    isCurrentLanguageRtl() {
        return this.isRtl(this.getCurrentLanguage());
    }

    isCurrentLanguageChinese() {
        let l = this.getCurrentLanguage();
        return ['zhs'].indexOf(l) >= 0;
    }

    isRtl(language: string) {
        if (language === 'he' || language === 'ar') {
            return true;
        }
        return false;
    }

    getLanguageFull(language: string, separator = '_') {
        switch (language) {
            case 'us':
                return 'en' + separator + 'US';
            case 'en':
                return 'en' + separator + 'GB';
            case 'ua':
                return 'uk' + separator + 'UA';
            case 'zhs':
                return 'zh' + separator + 'CN';
            case 'zht':
                return 'zh' + separator + 'TW';
            case 'ae':
                return 'ar' + separator + 'AE';
            default:
                return language + separator + language.toUpperCase();
        }
    }
}

@Injectable()
export class LogMissingTranslationHandler implements MissingTranslationHandler {
    handle(params: MissingTranslationHandlerParams) {
        return params.key;
    }
}

// @Injectable()
// export class CustomTranslateLoader implements TranslateLoader {
//     getLanguage(language: string) {
//         return Translate.loadFile(language);
//     }

//     getTranslation(lang: string): Observable<any> {
//         return new Observable<any>((observer: Observer<any>) => {
//             this.getLanguage(lang).then(val => {
//                 observer.next(val);
//                 observer.complete();
//             });
//         });
//     }
// }
