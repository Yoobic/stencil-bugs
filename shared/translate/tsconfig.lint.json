{
  "extends": "../../tsconfig.json",
  "compilerOptions": {
    "outDir": "../../out-tsc/spec",
    "module": "esnext",
    "types": []
  },
  "exclude": [
    "src/test.ts",
    "**/*.spec.ts"
  ]
}
