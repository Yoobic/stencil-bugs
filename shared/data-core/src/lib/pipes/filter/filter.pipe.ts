import { Pipe, PipeTransform } from '@angular/core';
import { Translate } from '@shared/translate';
import { isBlank } from '@shared/common';
import { keys as _keys } from 'lodash-es';

@Pipe({ name: 'filter' })
export class FilterPipe implements PipeTransform {

    private translate: Translate;

    constructor(translate: Translate) {
        this.translate = translate;
    }

    transform(value: Array<any>, ...args: any[]): any {
        //value: array of values to filter
        //args[0]: string to match
        //args[1]: list of fields to use to compare
        //args[2]: not sure? seems to exlude if false
        //args[3]: use translation

        if (!args || (!args[0])) { //&& !args[1]
            return value;
        } else if (value) {
            let translate = args[3];
            return value.filter(item => {
                if (typeof item === 'string') {
                    return item && this.getStringToTest(item, translate).toLowerCase().indexOf(args[0].toLowerCase()) >= 0;
                }
                let keys = args[1] || _keys(item);
                keys = [].concat(keys);

                let retVal = false;
                for (let i = 0; i < keys.length; i++) {
                    let key = keys[i];
                    if (args[0] && args[0].toLowerCase) {
                        if ((typeof item[key] === 'string' || item[key] instanceof String)) {
                            let index = this.getStringToTest(item[key], translate).toLowerCase().indexOf(args[0].toLowerCase());
                            retVal = retVal || (args[2] === false ? index < 0 : index >= 0);
                        }
                    } else if (!args[0] && args[1]) {
                        let bool = item.hasOwnProperty(key) && !isBlank(item[key]); //item.hasOwnProperty(key) && item[key];
                        retVal = retVal || (args[2] === false ? !bool : bool);
                    }
                }
                return retVal;
            });
        }
    }

    getStringToTest(s: string, translate: boolean): string {
        if (translate && this.translate) {
            return this.translate.get(s.toString().toUpperCase());
        } else {
            return s;
        }
    }
}