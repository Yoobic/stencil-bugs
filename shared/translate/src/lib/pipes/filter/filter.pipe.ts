import { Pipe, PipeTransform } from '@angular/core';
import { Translate } from '../../services/translate/translate.service';
import { isBlank } from '@shared/common';
import { keys as _keys } from 'lodash-es';

@Pipe({ name: 'filter' })
export class FilterPipe implements PipeTransform {

    private translate: Translate;

    constructor(translate: Translate) {
        this.translate = translate;
    }

    transform(value: Array<any>, ...args: any[]): any {
        if (!args || (!args[0])) {
            return value;
        } else if (value) {
            const translate = args[3];
            return value.filter(item => {
                if (typeof item === 'string') {
                    return item && this.getStringToTest(item, translate).toLowerCase().indexOf(args[0].toLowerCase()) >= 0;
                }
                let keys = args[1] || _keys(item);
                keys = [].concat(keys);

                let retVal = false;
                for (let i = 0; i < keys.length; i++) {
                    const key = keys[i];
                    if (args[0] && args[0].toLowerCase) {
                        if ((typeof item[key] === 'string' || item[key] instanceof String)) {
                            let index = this.getStringToTest(item[key], translate).toLowerCase().indexOf(args[0].toLowerCase());
                            retVal = retVal || (args[2] === false ? index < 0 : index >= 0);
                        }
                    } else if (!args[0] && args[1]) {
                        let bool = item.hasOwnProperty(key) && !isBlank(item[key]);
                        retVal = retVal || (args[2] === false ? !bool : bool);
                    }
                }
                return retVal;
            });
        }
    }

    getStringToTest(s: string, translate: boolean): string {
        if (translate && this.translate) {
            return this.translate.get(s.toString().toUpperCase());
        } else {
            return s;
        }
    }
}