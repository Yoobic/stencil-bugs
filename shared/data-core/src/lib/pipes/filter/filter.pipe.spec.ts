import { FilterPipe } from './filter.pipe';

import { Translate } from '@shared/translate';

describe('pipes: filterPipe', () => {

    const DATA = [
        { first: 'john', last: 'Doe', status: 'HAPPY_LNG' },
        { first: 'Jane', last: 'Doe', status: 'HAPPY_LNG' },
        { first: 'Jill', last: 'Day', status: 'SAD_LNG' }
    ];

    const LNG = {
        HAPPY_LNG: 'Happy',
        SAD_LNG: 'Sad'
    };

    let filterPipe: FilterPipe;
    let mockTranslate = new Translate(null, null, null);

    beforeEach(() => {
        filterPipe = new FilterPipe(mockTranslate);
    });
    it('should return data if no string to match', () => {
        expect(filterPipe.transform(DATA)).toMatchSnapshot();
    });

    it('should filter data by value', () => {
        expect(filterPipe.transform(DATA, 'john')).toMatchSnapshot();
    });

    it('should be case insensitive', () => {
        expect(filterPipe.transform(DATA, 'JOHN')).toMatchSnapshot();
    });
    it('should limit filter to certain fileds', () => {
        expect(filterPipe.transform(DATA, 'john', ['last', 'status'])).toEqual([]);
    });

    it('should reverse filter', () => {
        expect(filterPipe.transform(DATA, 'john', ['first'], false)).toMatchSnapshot();
    });

    it('should use translated value', () => {
        mockTranslate.get = jest.fn(key => LNG[key] || '');
        expect(filterPipe.transform(DATA, 'Happy', ['status'], true, true)).toMatchSnapshot();
    });

});
