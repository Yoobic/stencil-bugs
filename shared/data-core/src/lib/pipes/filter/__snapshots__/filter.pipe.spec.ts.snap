// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`pipes: filterPipe should be case insensitive 1`] = `
Array [
  Object {
    "first": "john",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
]
`;

exports[`pipes: filterPipe should filter data by value 1`] = `
Array [
  Object {
    "first": "john",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
]
`;

exports[`pipes: filterPipe should return data if no string to match 1`] = `
Array [
  Object {
    "first": "john",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
  Object {
    "first": "Jane",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
  Object {
    "first": "Jill",
    "last": "Day",
    "status": "SAD_LNG",
  },
]
`;

exports[`pipes: filterPipe should reverse filter 1`] = `
Array [
  Object {
    "first": "Jane",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
  Object {
    "first": "Jill",
    "last": "Day",
    "status": "SAD_LNG",
  },
]
`;

exports[`pipes: filterPipe should use translated value 1`] = `
Array [
  Object {
    "first": "john",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
  Object {
    "first": "Jane",
    "last": "Doe",
    "status": "HAPPY_LNG",
  },
]
`;
